[IMP] inventory: invoice shipping (#8325)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping/operation/invoicing.rst

In Odoo, shipping costs can be invoiced in two ways:

#. :ref:`Agree with the customer on a fixed cost and include it in the sale order.
   <inventory/shipping/invoice-so>`

#. Invoice shipping costs to the customer post-delivery, reflecting the actual expenses incurred by
   the business.

Configuration
=============

To set prices to delivery methods, go to :menuselection:`Inventory app --> Configuration -->
Settings`. Under the :guilabel:`Shipping` section, enable the :guilabel:`Delivery Methods` feature.
Then, click :guilabel:`Save`.

.. image:: invoicing/enable-delivery.png
   :align: center
   :alt: Enable the "Delivery Methods" feature in Settings.

Add shipping method
===================

Next, configure the the price of each delivery method by going to :menuselection:`Inventory app -->
Configuration --> Shipping Methods` and click the :guilabel:`Create` button. Doing so opens a form
to provide details about the shipping provider, including:

- :guilabel:`Shipping Method` (*Required field*): the name of the delivery method (e.g. `flat-rate
  shipping`, `same day delivery`, etc.).
- :guilabel:`Provider` (*Required field*): choose the delivery service, like Fedex, if using a

>   :doc:` third-party carrier <../setup/third_party_shipper>` Ensure the integration with the

Author: i added an extra space! fixed

>   shipping carrier is properly installed and select the provider from the drop-down menu.

Author: added an admonition block to link to the third party integration

- :guilabel:`Delivery Product` (*Required field*): the product listed on the :ref:`sales order line
  <inventory/shipping/sales-order>` as the delivery charge.
- :guilabel:`Free if order amount is above`: checking this box enables free shipping if the customer
  spends above the specified amount.

.. _inventory/shipping/invoice-so:

Invoice cost on sales order
===========================

To invoice shipping costs on the sales order, before the item is delivered, go to the
:menuselection:`Sales app` and select the desired sales order.

On the sales order, click the :guilabel:`Add Shipping` button at the bottom-right corner.

.. image:: invoicing/add-shipping.png
   :align: center
   :alt: Click "Add Shipping" button at the bottom right, near the total.

In the :guilabel:`Add a shipping method` pop-up window, choose the intended carrier in the
:guilabel:`Shipping Method` field.

Then, click the :guilabel:`Get Rate` button to the calculate shipping price based on real-time
shipping data Odoo's shipping carrier integration.

The :guilabel:`Cost` is automatically calculated using the weight of the items in the order.
Finally, click the :guilabel:`Add` button to close the window.

> .. image:: invoicing/add-a-shipping-method.png

Author: updated the image to address this feedback (:

   :align: center
   :alt: Calculate shipping by selecting a shipping method.

.. _inventory/shipping/sales-order:

On the sales order, the delivery product appears in the :guilabel:`Order Lines` tab, with the
:guilabel:`Unit Price` set as the shipping cost calculated in the :guilabel:`Add a shipping method`
pop-up window.

> .. image:: invoicing/delivery-product.png

Author: updated image to address feedback (:

   :align: center
   :alt: Show delivery product on the sales order line.

Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
that includes the shipping cost that was added earlier.

Invoice real shipping costs

> ===========================

Reviewer: Overall suggestion for this section is to add more UI walk-through. We *tell* more than *show* here and I'm confident this feedback would come up in KC's review anw 😉

To modify the invoice to reflect the real cost of shipping, follow the steps :ref:`above

> <inventory/shipping/invoice-so>` to create an invoice with a delivery product with a :guilabel:`Unit
> Price` of zero.

Reviewer: The steps above are about creating an SO, not an invoice, which seems to be what we'd want to do here first.
```suggestion
<inventory/shipping/invoice-so>` to create and and confirm a sales order with a delivery product with a :guilabel:`Unit
Price` of zero.
```

Author: I'm not sure if I agree? The steps above take a longer time adding the delivery product on the SO, yes, but the last instruction is to create an invoice. In line 88:
> Finally, after the product is delivered, click :guilabel:`Create invoice`, and an invoice is created
that includes the shipping cost that was added earlier.
